### PR TITLE
[C++ API] C++ API doc update

### LINKED ISSUE #16706: [C++ API] documents
- Add the documents which Java API already has.
image

### PR DESCRIPTION
**Pull request by SongGuyang**
## Why are these changes needed?
- We already support build C++ application from pip package.

image

### CODE REVIEW

File: doc/source/index.rst [hunk 1]
```diff
@@ -116,30 +116,29 @@ Ray provides Python, Java, and *EXPERIMENTAL* C++ API. And Ray uses Tasks (funct
...
-    | Option 2: connect to an existing Ray cluster with a known redis address (e.g. `127.0.0.1:6379`).
+    | - Option 2: connect to an existing Ray cluster with a known redis address (e.g. `127.0.0.1:6379`).
 
     .. code-block:: shell
 
-      ray stop
       ray start --head
       RAY_ADDRESS=127.0.0.1:6379 ./bazel-bin/example
 
+    | - Now, you can start to modify example.cc and build your ray C++ application.
```
Comment (qicosmos) [inline doc/source/index.rst:140]: start or modify?

File: doc/source/index.rst [hunk 2]
```diff
@@ -116,30 +116,29 @@ Ray provides Python, Java, and *EXPERIMENTAL* C++ API. And Ray uses Tasks (funct
...
-      ./bazel-bin/example 
+      cd ray-template && bazel run //:example
 
-    | Option 2: connect to an existing Ray cluster with a known redis address (e.g. `127.0.0.1:6379`).
+    | - Option 2: connect to an existing Ray cluster with a known redis address (e.g. `127.0.0.1:6379`).
 
     .. code-block:: shell
 
-      ray stop
       ray start --head
```
Comment (qicosmos) [inline doc/source/index.rst:137]: "RAY_ADDRESS=127.0.0.1:6379 ./bazel-bin/example" is in another line? or in the same line?

File: doc/source/index.rst [hunk 3]
```diff
@@ -116,30 +116,29 @@ Ray provides Python, Java, and *EXPERIMENTAL* C++ API. And Ray uses Tasks (funct
...
+
+    | - Install ray with C++ API support and generate bazel project template by ray script.
 
     .. code-block:: shell
 
-      pip install -e python --verbose
-      cd cpp/example/
-      bazel build //:example
+      pip install ray[cpp]
+      mkdir ray-template && ray cpp --generate-bazel-project-template-to ray-template
```
Comment (qicosmos) [inline doc/source/index.rst:125]: if pip install ray cpp, do we still have to mkdir ray-template?

File: doc/source/index.rst [hunk 4]
```diff
@@ -116,30 +116,29 @@ Ray provides Python, Java, and *EXPERIMENTAL* C++ API. And Ray uses Tasks (funct
...
 
-    | Option 1: run the example directly with a dynamic library path. It will start a Ray cluster automatically.
+    | - Option 1: run the example directly. It will start a Ray cluster automatically.
```
Comment (raulchen) [inline doc/source/index.rst:127]: The project template comes with a simple example application. You can try this example out in 2 ways:
1. Run the example application directly, which will start a Ray cluster locally.
2. Connect the example application to an existing Ray cluster by specifying the `RAY_ADDRESS` env var.

File: doc/source/index.rst [hunk 5]
```diff
@@ -116,30 +116,29 @@ Ray provides Python, Java, and *EXPERIMENTAL* C++ API. And Ray uses Tasks (funct
 
     | The C++ Ray API is currently experimental with limited support. You can track its development `here <https://github.com/ray-project/ray/milestone/17>`__ and report issues on GitHub.
     | Run the following commands to get started:
-    | - Build ray from source with *bazel* as shown `here <https://docs.ray.io/en/master/development.html#building-ray-full>`__.
-    | - Modify and build `cpp/example/example.cc`.
+
+    | - Install ray with C++ API support and generate bazel project template by ray script.
```
Comment (raulchen) [inline doc/source/index.rst:120]: Install ray with C++ API support and generate a bazel project with the `ray` command.

**Review by raulchen — APPROVED** [review]

File: doc/source/index.rst (hunk 1 above)
Comment (raulchen) [reply doc/source/index.rst:140]: Now you can build your own Ray C++ application based on this project template.

File: doc/source/index.rst [hunk 6]
```diff
@@ -116,30 +116,29 @@ Ray provides Python, Java, and *EXPERIMENTAL* C++ API. And Ray uses Tasks (funct
...
 
     .. code-block:: shell
 
-      ray stop
       ray start --head
       RAY_ADDRESS=127.0.0.1:6379 ./bazel-bin/example
 
+    | - Now, you can start to modify example.cc and build your ray C++ application.
+
     .. literalinclude:: ../../cpp/example/example.cc
```
Comment (raulchen) [inline doc/source/index.rst:142]: I don't think it's necessary to include the `example.cc` code content here.

File: doc/source/index.rst (hunk 2 above)
Comment (SongGuyang) [reply doc/source/index.rst:137]: You can see my screenshot. One command, one line.

**Pull request merged**